### PR TITLE
614 conn detail diff single line

### PR DESCRIPTION
**Pull request by ShiriMoran**

### CODE REVIEW

**Comment by ShiriMoran** [comment]
fixes #615

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/vpcmodel/jsonOutput.go [hunk 1]
```diff
@@ -52,12 +52,14 @@ type connLine struct {
...
-	Conn1     connection.Details `json:"conn1"`
-	Conn2     connection.Details `json:"conn2"`
+	SrcChange           string             `json:"src_change"`
+	DstChange           string             `json:"dst_change"`
+	Src                 EndpointElem       `json:"src"`
+	Dst                 EndpointElem       `json:"dst"`
+	Conn1               connection.Details `json:"conn1,omitempty"`
+	UnidirectionalConn1 connection.Details `json:"unidirectional_conn1,omitempty"`
+	Conn2               connection.Details `json:"conn2,omitempty"`
+	UnidirectionalConn2 connection.Details `json:"unidirectional_conn2,omitempty"`
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/vpcmodel/jsonOutput.go]: done

**Review by zivnevo — APPROVED** [review]

File: pkg/vpcmodel/semanticDiff.go [hunk 2]
```diff
@@ -192,16 +192,15 @@ func (confConnectivity *configConnectivity) connMissingOrChanged(other *configCo
 				return nil, err2
 			}
 			// includeChanged indicates if it is thisMinusOther
-			connDiff := &connectionDiff{connsResponsive, nil, missingConnection, includeChanged}
+			connDiff := &connectionDiff{conn, emptyDetailedConn(),
+				missingConnection, includeChanged}
```
Comment (adisos) [inline pkg/vpcmodel/semanticDiff.go:196]:
```suggestion
			connDiff := &connectionDiff{
				conn1:          conn,
				conn2:          emptyDetailedConn(),
				diff:           missingConnection,
				thisMinusOther: includeChanged,
			}
```

File: pkg/ibmvpc/analysis_output_test.go [hunk 3]
```diff
@@ -470,6 +470,11 @@ var tests = []*vpcGeneralTest{
 		useCases:    []vpcmodel.OutputUseCase{vpcmodel.EndpointsDiff},
 		format:      vpcmodel.Text,
 	},
+	{
+		inputConfig: "acl_testing3",
+		useCases:    []vpcmodel.OutputUseCase{vpcmodel.EndpointsDiff},
+		format:      vpcmodel.JSON,
+	},
```
Comment (adisos) [inline pkg/ibmvpc/analysis_output_test.go:477]: for the `diff` command we do not yet officially support the JSON format (I recall there could be some open issue with relevant questions about this output) .

File: pkg/vpcmodel/detailedConn.go [hunk 4]
```diff
@@ -109,14 +111,30 @@ func (d *detailedConn) nonTCPAndResponsiveTCPComponent() *connection.Set {
...
 
-// todo: will it still be needed once transformation is completed?
 func (d *detailedConn) string() string {
 	if !d.tcpRspDisable.IsEmpty() {
 		return d.allConn.String() + asterisk
 	}
 	return d.allConn.String()
 }
 
+func (d *detailedConn) detailString() string {
```
Comment (adisos) [inline pkg/vpcmodel/detailedConn.go:121]: I would consider renaming this function and adding some documentation, about the changes from the `string()` method.
some comment to indicate that the responsiveness indication is per cube, not per the entire connection string.

**Review by adisos — COMMENTED** [review]
Added few small comments.

File: pkg/ibmvpc/analysis_output_test.go (hunk 3 above)
Comment (ShiriMoran) [reply pkg/ibmvpc/analysis_output_test.go:477]: ok, removed.

File: pkg/vpcmodel/detailedConn.go (hunk 4 above)
Comment (ShiriMoran) [reply pkg/vpcmodel/detailedConn.go]: done

File: pkg/ibmvpc/examples/out/analysis_out/acl_testing3_all_vpcs_endpointsDiff.json
Comment (adisos) [inline pkg/ibmvpc/examples/out/analysis_out/acl_testing3_all_vpcs_endpointsDiff.json:1]: this file should also be removed

Comment (ShiriMoran) [reply pkg/ibmvpc/examples/out/analysis_out/acl_testing3_all_vpcs_endpointsDiff.json:1]: right. I had it in mind and somehow forget. fone

**Review by adisos — APPROVED** [review]

File: pkg/ibmvpc/analysis_output_test.go (hunk 3 above)
Comment (ShiriMoran) [reply pkg/ibmvpc/analysis_output_test.go]: I see there are in fact json testing @adisos 
```
	// json examples
	{
		inputConfig: "demo_with_instances",
		useCases:    []vpcmodel.OutputUseCase{vpcmodel.AllEndpoints, vpcmodel.AllSubnets},
		format:      vpcmodel.JSON,
	},
	{
		inputConfig: "acl_testing3",
		useCases:    []vpcmodel.OutputUseCase{vpcmodel.AllEndpoints},
		format:      vpcmodel.JSON,
	},
```

**Pull request merged**